### PR TITLE
types: Refactor EventAttribute

### PR DESCRIPTION
**Pull request by alexanderbez**
closes: #6403

### CODE REVIEW

File: proto/tendermint/abci/types.proto [hunk 1]
```diff
@@ -275,17 +269,16 @@ message LastCommitInfo {
...
 // EventAttribute is a single key-value pair, associated with an event.
 message EventAttribute {
-  bytes key   = 1;
-  bytes value = 2;
-  bool  index = 3;  // nondeterministic
+  // bytes key   = 1;
+  // bytes value = 2;
+  string key   = 1;
+  string value = 2;
+  bool   index = 3;  // nondeterministic
```
Comment (alexanderbez) [inline proto/tendermint/abci/types.proto:281]: @ethanfrey should we break proto here by changing the types or should we introduce new types, `key_str` and `value_str`, and use those while keeping the existing ones deprecated?

/cc @tychoish

Comment (ethanfrey) [reply proto/tendermint/abci/types.proto:281]: Good question. Responded to this in the issue with my opinion. Curious as to the implications of the queries. Or binary.

If bytes field held utf8 string before, then these two are identical. It only breaks when illegal binary data was present

Comment (alexanderbez) [reply proto/tendermint/abci/types.proto:281]: Yeah I think this change is fine. I'll wrap this up then.

**Review by marbar3778 — APPROVED** [review]
LGTM.

Are we fine with this breaking change for 0.35?

**Comment by ethanfrey** [comment]
FYI it is less breaking than it appears.

https://developers.google.com/protocol-buffers/docs/proto#updating

> string and bytes are compatible as long as the bytes are valid UTF-8.

So anyone who didn't have trouble earlier (passing in utf8) will not even notice the change and can use 0.35 without updating their protobuf defs abci side

**Pull request merged**